### PR TITLE
Add staticcheck static analyzer to Travis

### PR DESCRIPTION
**Pull request by y0ssar1an**
[`staticcheck`](https://github.com/dominikh/go-staticcheck) is a popular static analyzer. It's a free bug finder that catches some things that pass `go vet`.

### CODE REVIEW

File: test/bench_test.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ func BenchmarkPubSubSpeed(b *testing.B) {
 		}
 		// Don't overrun ourselves and be a slow consumer, server will cut us off
 		if int32(i)-atomic.LoadInt32(&received) > 32768 {
-			time.Sleep(100)
+			time.Sleep(100 * time.Nanosecond)
```
Comment (y0ssar1an) [inline test/bench_test.go]: I'm not sure if the test author intended to sleep for 100ns, but it's best to be explicit about the duration.

**Comment by y0ssar1an** [comment]
Can't make `staticcheck` work with Travis :(

**Comment by kozlovic** [comment]
Are you sure you need the `go_import_path: honnef.co/go/staticcheck/cmd/staticcheck` line? Is `go get` not enough?

**Comment by kozlovic** [comment]
Experimented with removing the `go_import_path` statement and it seems to be working now. Check the branch: https://github.com/nats-io/nats/tree/test_staticcheck

**Comment by y0ssar1an** [comment]
I initially tried without `go_import_path` and it didn't work. I now have an identical `.travis.yml` file in #233 to the one in `test_staticcheck` and it's still not working.

**Pull request closed without merging**